plugins/optimization-detective/site-health.php

```diff
@@ -0,0 +1,167 @@
+			'status_code'   => $status_code,
+			'available'     => false,
+			'error_message' => '',
+		);
+
+		if (
+			400 === $status_code &&
+			isset( $data['data']['params'] ) &&
+			is_array( $data['data']['params'] ) &&
+			count( $data['data']['params'] ) > 0
+		) {
+			// The REST API endpoint is available.
+			$info['available'] = true;
+		} elseif ( 401 === $status_code ) {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'Authorization should not be required to access the Optimization Detective REST API endpoint.', 'optimization-detective' );
+			$result['description'] = esc_html__( 'To collect URL metrics, the Optimization Detective REST API endpoint should be accessible without requiring authorization.', 'optimization-detective' );
+		} elseif ( 403 === $status_code ) {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'Access to the Optimization Detective REST API endpoint was denied.', 'optimization-detective' );
+			$result['description'] = esc_html__( 'Access was denied because the user does not have the necessary capabilities. Please check user roles and capabilities, as all users should have access to the Optimization Detective REST API endpoint.', 'optimization-detective' );
+		} else {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'Error accessing the Optimization Detective REST API endpoint', 'optimization-detective' );
+			$result['description'] = esc_html__( 'There was an issue reaching the Optimization Detective REST API endpoint. This might be due to server settings or the REST API being disabled.', 'optimization-detective' );
+		}
+		$info['error_message'] = $result['description'];
+	}
+
+	$result['description'] = sprintf(
+		'<p>%s</p>',
+		$result['description']
+	);
+
+	update_option( 'od_rest_api_info', $info );
+	return $result;
+}
+
+/**
+ * Renders an admin notice if the REST API health check fails.
+ *
+ * @since n.e.x.t
+ *
+ * @param array<string> $additional_classes Additional classes to add to the notice.
+ */
+function od_render_rest_api_health_check_notice( array $additional_classes ): void {
+	$rest_api_info = get_option( 'od_rest_api_info', array() );
+	if (
+		isset( $rest_api_info['available'] ) &&
+		false === $rest_api_info['available'] &&
+		isset( $rest_api_info['error_message'] ) &&
+		'' !== $rest_api_info['error_message']
+	) {
+		wp_admin_notice(
+			esc_html( $rest_api_info['error_message'] ),
+			array(
+				'type'               => 'warning',
+				'additional_classes' => $additional_classes,
+			)
+		);
+	}
+}
+
+/**
+ * Displays an admin notice on the plugin row if the REST API health check fails.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $plugin_file Plugin file.
+ */
+function od_rest_api_health_check_admin_notice( string $plugin_file ): void {
+	if ( 'optimization-detective/load.php' !== $plugin_file ) {
+		return;
+	}
+	od_render_rest_api_health_check_notice( array( 'inline', 'notice-alt' ) );
```

Comment:
Note the outstanding comment in regards to using `inline`: https://github.com/WordPress/performance/pull/1762#discussion_r1907965873

Curious, what does the `inline` class actually do? Why should we or should we not use it here?

cc @westonruter

Reply:
The `inline` class prevents the notice from being moved up to the admin notices area at the top of the screen.

I think there should be one notice always displayed in the plugin row with `inline`, and then another admin notice displayed at the top once after activating the plugin. Upon reloading the page, the notice at the top would go away but the notice shown in the plugin row would be persistent. This would increase visibility of something which would otherwise only be discovered in Site Health.

Reply:
So in that case we would need it here, correct? And the other call to render the notice would cater for the overall notice I assume?

Reply:
Text refined in 4c1b2dc35e1d64951f2bb33b35cda7ee971a8cab